### ISSUE TITLE
Unable to create PVCs as ReadOnlyMany with 1.4.0

### ISSUE THREAD

**Issue by bruce2409** [kubernetes-sigs/gcp-compute-persistent-disk-csi-driver#928]
I've recently deployed a new GKE cluster which is running version 1.4.0 of the provided CSI driver and it seems I'm not able to create PVC with the access mode set to ReadOnlyMany.

This just leads to the error: `failed to provision volume with StorageClass "gke-balanced": rpc error: code = InvalidArgument desc = VolumeContentSource must be provided when AccessMode is set to read only`. No PV is created and the PVC stays stuck in pending. 

ReadWriteOnce volumes seem to work just fine, but I noticed that the pending PVC doesn't create any entries in the CSI pod logs, thought the RWO volumes do.

On another cluster which is still running 1.3.4 the exact same PVC definition works properly with ROX.

My StorageClass is:
```
allowVolumeExpansion: true
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  annotations:
    storageclass.kubernetes.io/is-default-class: "true"
  name: gke-balanced
parameters:
  type: pd-balanced
provisioner: pd.csi.storage.gke.io
reclaimPolicy: Delete
volumeBindingMode: Immediate
```

**Comment by mattcary** [comment]
Creating disks with access mode read-only many never actually worked in the PD layer: the disk was not actually created as read-only, and so would not be able to be attached to multiple nodes (which is a key use case for read-only disks). This was fixed in 1.4.0.

The right flow for read-only is described here: https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/readonlymany-disks.

Since the driver now correctly creates disks as read-only, they must be created from a snapshot or clone. They can't be mounted writable and populated, because the underlying PD is now actually read-only.

**Comment by ellamacrow98** [comment]
Hello, I want to add additional software to the live system. But after that, I want to use it like a normal live system (only readable) and don't want to store any temporary data on this drive.

I thought it might be somehow possible by creating a persistent volume and starting loading it into a ramdisk.
Do someone know if something like this is possible with any Linux live system?
